src/txBaseMsgProcessor.cpp

```diff
@@ -0,0 +1,444 @@
+#include <txBaseMsgProcessor.h>
+#include <sc/proofverifier.h>
+#include <main.h>
+#include <util.h>
+#include <consensus/validation.h>
+
+#include <set>
+
+TxBaseMsgProcessor::TxBaseMsg_DataToProcess::TxBaseMsg_DataToProcess():
+    txBaseHash(), sourceNodeId(-1),
+    pTxBase(nullptr), pSourceNode(nullptr),
+    txBaseProcessingState(MempoolReturnValue::NOT_PROCESSED_YET), txBaseValidationState() {};
+
+TxBaseMsgProcessor::TxBaseMsgProcessor() { this->reset(); }
+TxBaseMsgProcessor::~TxBaseMsgProcessor() { this->reset(); }
+
+void TxBaseMsgProcessor::reset()
+{
+    {
+        LOCK(cs_rejectedTxes);
+        recentRejects.reset(nullptr);
+        recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
```

Comment:
It seems like the first reset (with _nullptr_) is not necessary.

```suggestion
        recentRejects.reset(new CRollingBloomFilter(120000, 0.000001));
```